elasticsearch/elasticsearch.d.ts

```diff
@@ -0,0 +1,253 @@
+// Type definitions for elasticsearch
+// Project: https://www.npmjs.com/package/elasticsearch
+// Definitions by: Casper Skydt <https://github.com/CasperSkydt/DefinitelyTyped>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module Elasticsearch {
+    export class Client {
+        constructor(params: any);
```

Comment:
looks like there are a few options here. from https://github.com/elastic/elasticsearch-js/blob/master/src/lib/client.js, a better signature would be 
```ts
constructor(config: {log?: "error" | "warning" | "info" | "debug" | "trace", host?:string, sniffEndpoint?: string, plugins?: Function[] })
```